sycl/doc/extensions/proposed/sycl_ext_oneapi_syclbin.asciidoc

```diff
@@ -0,0 +1,244 @@
+contents, which must in turn match the state of any `kernel_bundle` object
+created from it.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_SYCLBIN` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+
+=== New kernel_bundle interfaces
+
+|====
+a|
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template<bundle_state State, typename PropertyListT = empty_properties_t>
+kernel_bundle<State> get_kernel_bundle(const context& ctxt,
+                                       const std::vector<device>& devs,
+                                       const sycl::span<char>& bytes,
+                                       PropertyListT props = {});
+
+// Requires C++20
+template<bundle_state State, typename PropertyListT = empty_properties_t>
+kernel_bundle<State> get_kernel_bundle(const context& ctxt,
+                                       const std::vector<device>& devs,
+                                       const std::span<char>& bytes,
+                                       PropertyListT props = {});
+
+}
+----
+!====
+
+_Constraints:_ Available only when `State` is not `ext_oneapi_source`.
+
+_Effects:_ Creates a new kernel bundle containing the kernels from the SYCLBIN
+data contained in `bytes` that are compatible with at least one of the devices
+in `devs`. Any remaining kernels (those that are not compatible with any of the
+devices in `devs`) are not represented in the new kernel bundle.
+
+The bundle is associated with the context `ctxt`, and kernels from this bundle
+may only be submitted to a queue that shares the same context and whose device
+is in `devs`.
+
+_Returns:_ The newly created kernel bundle, which has `State` state.
+
+_Throws:_
+
+* An `exception` with the `errc::invalid` error code if the contents of `bytes`
+  is not in the SYCLBIN format, as defined by the SYCL implementation.
+* An `exception` with the `errc::invalid` error code if the SYCLBIN read from
+  `bytes` is not in the `State` state.
+* An `exception` with the `errc::invalid` error code if the `devs` vector is
+  empty.
+* An `exception` with the `errc::invalid` error code if `State` is
+  `bundle_state::input` and any device in `ctxt.get_devices()` does not have
+  `aspect::online_compiler`.
+* An `exception` with the `errc::invalid` error code if `State` is
+  `bundle_state::object` and any device in `ctxt.get_devices()` does not have
+  `aspect::online_linker`.
+* An `exception` with the `errc::build` error code if `State` is
+  `bundle_state::object` or `bundle_state::executable`, if the implementation
```

Comment:
Should those states be `input` or `object`? `executable` should not require any further processing